File: data_db/src/main/java/br/org/restaurantepopular/data_db/mapper/CategoryEntityMapper.java
package br.org.restaurantepopular.data_db.mapper;

import br.org.restaurantepopular.data_db.entity.CategoryEntity;
import br.org.restaurantepopular.entity.Category;

public class CategoryEntityMapper {

    public static CategoryEntity toCategoryEntity(Category category) {
        return new CategoryEntity.Builder()
                .id(category.getId())
                .name(category.getName())
                .description(category.getDescription())
                .build();
    }

    public static Category toCategory(CategoryEntity categoryEntity) {
        return new Category.CategoryBuilder()
                .id(categoryEntity.getId())
                .name(categoryEntity.getName())
                .description(categoryEntity.getDescription())
                .build();

    }
}

File: data_db/src/main/java/br/org/restaurantepopular/data_db/mapper/CategoryEntityMapper.kt
package br.org.restaurantepopular.data_db.mapper

import br.org.restaurantepopular.data_db.entity.CategoryEntity
import br.org.restaurantepopular.entity.Category

object CategoryEntityMapper {

    fun toCategoryEntity(category: Category): CategoryEntity {
        return CategoryEntity(
            id = category.id ?: "",
            name = category.name,
            description = category.description
        )
    }

    fun toCategory(categoryEntity: CategoryEntity): Category {
        return Category(
            id = categoryEntity.id,
            name = categoryEntity.name,
            description = categoryEntity.description
        )
    }
}